### ISSUE TITLE
NullPointerException when tries to send an error to a client (subscriptions and queries via websockets)

### ISSUE THREAD

**Issue by yatagan** [walmartlabs/lacinia#153]
If I use websockets as a transport, a client does not receive query validation errors. Server throws exceptions like this:

```
Exception in thread "async-dispatch-23" clojure.lang.ExceptionInfo: java.lang.NullPointerException in Interceptor :com.walmartlabs.lacinia.pedestal.subscriptions/exception-handler -  {:execution-id 26, :stage :error, :interceptor :com.walmartlabs.lacinia.pedestal.subscriptions/exception-handler, :exception-type :java.lang.NullPointerException, :exception #error {
 :cause nil
 :via
 [{:type java.lang.NullPointerException
   :message nil
   :at [clojure.string$trim invokeStatic "string.clj" 238]}]
 :trace
 [[clojure.string$trim invokeStatic "string.clj" 238]
  [clojure.string$trim invoke "string.clj" 234]
  [com.walmartlabs.lacinia.pedestal.subscriptions$fix_up_message invokeStatic "subscriptions.clj" 156]
  [com.walmartlabs.lacinia.pedestal.subscriptions$fix_up_message invoke "subscriptions.clj" 152]
  [clojure.core$map$fn__4785 invoke "core.clj" 2646]
  [clojure.lang.LazySeq sval "LazySeq.java" 40]
  [clojure.lang.LazySeq seq "LazySeq.java" 49]
  [clojure.lang.LazySeq first "LazySeq.java" 71]
  [clojure.lang.RT first "RT.java" 667]
  [clojure.core$first__4339 invokeStatic "core.clj" 55]
  [clojure.string$join invokeStatic "string.clj" 180]
  [clojure.string$join invoke "string.clj" 180]
  [com.walmartlabs.lacinia.pedestal.subscriptions$construct_exception_message invokeStatic "subscriptions.clj" 181]
  [com.walmartlabs.lacinia.pedestal.subscriptions$construct_exception_message invoke "subscriptions.clj" 159]
  [com.walmartlabs.lacinia.pedestal.subscriptions$fn__36779 invokeStatic "subscriptions.clj" 192]
  [com.walmartlabs.lacinia.pedestal.subscriptions$fn__36779 invoke "subscriptions.clj" 188]
  [io.pedestal.interceptor.chain$try_error invokeStatic "chain.clj" 75]
  [io.pedestal.interceptor.chain$try_error invoke "chain.clj" 64]
  [io.pedestal.interceptor.chain$leave_all_with_binding invokeStatic "chain.clj" 253]
  [io.pedestal.interceptor.chain$leave_all_with_binding invoke "chain.clj" 237]
  [io.pedestal.interceptor.chain$leave_all$fn__31907 invoke "chain.clj" 268]
  [clojure.lang.AFn applyToHelper "AFn.java" 152]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.core$apply invokeStatic "core.clj" 646]
  [clojure.core$with_bindings_STAR_ invokeStatic "core.clj" 1881]
  [clojure.core$with_bindings_STAR_ doInvoke "core.clj" 1881]
  [clojure.lang.RestFn invoke "RestFn.java" 425]
  [io.pedestal.interceptor.chain$leave_all invokeStatic "chain.clj" 266]
  [io.pedestal.interceptor.chain$leave_all invoke "chain.clj" 260]
  [io.pedestal.interceptor.chain$execute invokeStatic "chain.clj" 379]
  [io.pedestal.interceptor.chain$execute invoke "chain.clj" 352]
  [com.walmartlabs.lacinia.pedestal.subscriptions$execute_query_interceptors invokeStatic "subscriptions.clj" 78]
  [com.walmartlabs.lacinia.pedestal.subscriptions$execute_query_interceptors invoke "subscriptions.clj" 56]
  [com.walmartlabs.lacinia.pedestal.subscriptions$connection_loop$fn__36575$state_machine__5576__auto____36586$fn__36597 invoke "subscriptions.clj" 90]
  [com.walmartlabs.lacinia.pedestal.subscriptions$connection_loop$fn__36575$state_machine__5576__auto____36586 invoke "subscriptions.clj" 90]
  [clojure.core.async.impl.ioc_macros$run_state_machine invokeStatic "ioc_macros.clj" 973]
  [clojure.core.async.impl.ioc_macros$run_state_machine invoke "ioc_macros.clj" 972]
  [clojure.core.async.impl.ioc_macros$run_state_machine_wrapped invokeStatic "ioc_macros.clj" 977]
  [clojure.core.async.impl.ioc_macros$run_state_machine_wrapped invoke "ioc_macros.clj" 975]
  [clojure.core.async$ioc_alts_BANG_$fn__5779 invoke "async.clj" 384]
  [clojure.core.async$do_alts$fn__5725$fn__5728 invoke "async.clj" 253]
  [clojure.core.async.impl.channels.ManyToManyChannel$fn__447$fn__448 invoke "channels.clj" 95]
  [clojure.lang.AFn run "AFn.java" 22]
  [java.util.concurrent.ThreadPoolExecutor runWorker "ThreadPoolExecutor.java" 1142]
  [java.util.concurrent.ThreadPoolExecutor$Worker run "ThreadPoolExecutor.java" 617]
  [java.lang.Thread run "Thread.java" 745]]}}
        at clojure.core$ex_info.invokeStatic(core.clj:4617)
        at clojure.core$ex_info.invoke(core.clj:4617)
        at io.pedestal.interceptor.chain$throwable__GT_ex_info.invokeStatic(chain.clj:35)
        at io.pedestal.interceptor.chain$throwable__GT_ex_info.invoke(chain.clj:32)
        at io.pedestal.interceptor.chain$try_error.invokeStatic(chain.clj:82)
        at io.pedestal.interceptor.chain$try_error.invoke(chain.clj:64)
        at io.pedestal.interceptor.chain$leave_all_with_binding.invokeStatic(chain.clj:253)
        at io.pedestal.interceptor.chain$leave_all_with_binding.invoke(chain.clj:237)
        at io.pedestal.interceptor.chain$leave_all$fn__31907.invoke(chain.clj:268)
        at clojure.lang.AFn.applyToHelper(AFn.java:152)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.core$apply.invokeStatic(core.clj:646)
        at clojure.core$with_bindings_STAR_.invokeStatic(core.clj:1881)
        at clojure.core$with_bindings_STAR_.doInvoke(core.clj:1881)
        at clojure.lang.RestFn.invoke(RestFn.java:425)
        at io.pedestal.interceptor.chain$leave_all.invokeStatic(chain.clj:266)
        at io.pedestal.interceptor.chain$leave_all.invoke(chain.clj:260)
        at io.pedestal.interceptor.chain$execute.invokeStatic(chain.clj:379)
        at io.pedestal.interceptor.chain$execute.invoke(chain.clj:352)
        at com.walmartlabs.lacinia.pedestal.subscriptions$execute_query_interceptors.invokeStatic(subscriptions.clj:78)
        at com.walmartlabs.lacinia.pedestal.subscriptions$execute_query_interceptors.invoke(subscriptions.clj:56)
        at com.walmartlabs.lacinia.pedestal.subscriptions$connection_loop$fn__36575$state_machine__5576__auto____36586$fn__36597.invoke(subscriptions.clj:90)
        at com.walmartlabs.lacinia.pedestal.subscriptions$connection_loop$fn__36575$state_machine__5576__auto____36586.invoke(subscriptions.clj:90)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:973)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:972)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:977)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:975)
        at clojure.core.async$ioc_alts_BANG_$fn__5779.invoke(async.clj:384)
        at clojure.core.async$do_alts$fn__5725$fn__5728.invoke(async.clj:253)
        at clojure.core.async.impl.channels.ManyToManyChannel$fn__447$fn__448.invoke(channels.clj:95)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
        at clojure.string$trim.invokeStatic(string.clj:238)
        at clojure.string$trim.invoke(string.clj:234)
        at com.walmartlabs.lacinia.pedestal.subscriptions$fix_up_message.invokeStatic(subscriptions.clj:156)
        at com.walmartlabs.lacinia.pedestal.subscriptions$fix_up_message.invoke(subscriptions.clj:152)
        at clojure.core$map$fn__4785.invoke(core.clj:2646)
        at clojure.lang.LazySeq.sval(LazySeq.java:40)
        at clojure.lang.LazySeq.seq(LazySeq.java:49)
        at clojure.lang.LazySeq.first(LazySeq.java:71)
        at clojure.lang.RT.first(RT.java:667)
        at clojure.core$first__4339.invokeStatic(core.clj:55)
        at clojure.string$join.invokeStatic(string.clj:180)
        at clojure.string$join.invoke(string.clj:180)
        at com.walmartlabs.lacinia.pedestal.subscriptions$construct_exception_message.invokeStatic(subscriptions.clj:181)
        at com.walmartlabs.lacinia.pedestal.subscriptions$construct_exception_message.invoke(subscriptions.clj:159)
        at com.walmartlabs.lacinia.pedestal.subscriptions$fn__36779.invokeStatic(subscriptions.clj:192)
        at com.walmartlabs.lacinia.pedestal.subscriptions$fn__36779.invoke(subscriptions.clj:188)
        at io.pedestal.interceptor.chain$try_error.invokeStatic(chain.clj:75)
        ... 29 more
```

In `debug` level logs I can see the validation error:

```
DEBUG io.pedestal.interceptor.chain - {:throw #error {
 :cause "Query validation errors."
 :data {:errors ({:message "Field `basement_levels' (of type `Value') must have at least one selection.", :locations [{:line 18, :column 17}]} {:message "Field `opts' (of type `FinOption') must have at least one selection.", :locations [{:line 18, :column 17}]})}
 :via
 [{:type clojure.lang.ExceptionInfo
   :message "Query validation errors."
   :data {:errors ({:message "Field `basement_levels' (of type `Value') must have at least one selection.", :locations [{:line 18, :column 17}]} {:message "Field `opts' (of type `FinOption') must have at least one selection.", :locations [{:line 18, :column 17}]})}
   :at [clojure.core$ex_info invokeStatic "core.clj" 4617]}]
 :trace
 [[clojure.core$ex_info invokeStatic "core.clj" 4617]
  [clojure.core$ex_info invoke "core.clj" 4617]
  [com.walmartlabs.lacinia.pedestal.subscriptions$query_parser_interceptor$fn__36790 invoke "subscriptions.clj" 234]
  [io.pedestal.interceptor.chain$try_f invokeStatic "chain.clj" 54]
  [io.pedestal.interceptor.chain$try_f invoke "chain.clj" 44]
  [io.pedestal.interceptor.chain$process_all_with_binding invokeStatic "chain.clj" 171]
  [io.pedestal.interceptor.chain$process_all_with_binding invoke "chain.clj" 146]
  [io.pedestal.interceptor.chain$process_all$fn__31880 invoke "chain.clj" 188]
  [clojure.lang.AFn applyToHelper "AFn.java" 152]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.core$apply invokeStatic "core.clj" 646]
  [clojure.core$with_bindings_STAR_ invokeStatic "core.clj" 1881]
  [clojure.core$with_bindings_STAR_ doInvoke "core.clj" 1881]
  [clojure.lang.RestFn invoke "RestFn.java" 425]
  [io.pedestal.interceptor.chain$process_all invokeStatic "chain.clj" 186]
  [io.pedestal.interceptor.chain$process_all invoke "chain.clj" 182]
  [io.pedestal.interceptor.chain$enter_all invokeStatic "chain.clj" 235]
  [io.pedestal.interceptor.chain$enter_all invoke "chain.clj" 229]
  [io.pedestal.interceptor.chain$execute invokeStatic "chain.clj" 379]
  [io.pedestal.interceptor.chain$execute invoke "chain.clj" 352]
  [com.walmartlabs.lacinia.pedestal.subscriptions$execute_query_interceptors invokeStatic "subscriptions.clj" 78]
  [com.walmartlabs.lacinia.pedestal.subscriptions$execute_query_interceptors invoke "subscriptions.clj" 56]
  [com.walmartlabs.lacinia.pedestal.subscriptions$connection_loop$fn__36575$state_machine__5576__auto____36586$fn__36597 invoke "subscriptions.clj" 90]
  [com.walmartlabs.lacinia.pedestal.subscriptions$connection_loop$fn__36575$state_machine__5576__auto____36586 invoke "subscriptions.clj" 90]
  [clojure.core.async.impl.ioc_macros$run_state_machine invokeStatic "ioc_macros.clj" 973]
  [clojure.core.async.impl.ioc_macros$run_state_machine invoke "ioc_macros.clj" 972]
  [clojure.core.async.impl.ioc_macros$run_state_machine_wrapped invokeStatic "ioc_macros.clj" 977]
  [clojure.core.async.impl.ioc_macros$run_state_machine_wrapped invoke "ioc_macros.clj" 975]
  [clojure.core.async$ioc_alts_BANG_$fn__5779 invoke "async.clj" 384]
  [clojure.core.async$do_alts$fn__5725$fn__5728 invoke "async.clj" 253]
  [clojure.core.async.impl.channels.ManyToManyChannel$fn__447$fn__448 invoke "channels.clj" 95]
  [clojure.lang.AFn run "AFn.java" 22]
  [java.util.concurrent.ThreadPoolExecutor runWorker "ThreadPoolExecutor.java" 1142]
  [java.util.concurrent.ThreadPoolExecutor$Worker run "ThreadPoolExecutor.java" 617]
  [java.lang.Thread run "Thread.java" 745]]}, :execution-id 12, :line 56}
DEBUG io.pedestal.interceptor.chain - {:in leave-all, :execution-id 12, :line 241}
DEBUG io.pedestal.interceptor.chain - {:interceptor :com.walmartlabs.lacinia.pedestal.subscriptions/exception-handler, :stage :error, :execution-id 12, :line 72}
DEBUG com.walmartlabs.lacinia.pedestal.subscriptions - {:event :com.walmartlabs.lacinia.pedestal.subscriptions/exception, :line 189}
clojure.lang.ExceptionInfo: clojure.lang.ExceptionInfo in Interceptor :com.walmartlabs.lacinia.pedestal.subscriptions/query-parser - Query validation errors.
        at clojure.core$ex_info.invokeStatic(core.clj:4617)
        at clojure.core$ex_info.invoke(core.clj:4617)
        at io.pedestal.interceptor.chain$throwable__GT_ex_info.invokeStatic(chain.clj:35)
        at io.pedestal.interceptor.chain$throwable__GT_ex_info.invoke(chain.clj:32)
        at io.pedestal.interceptor.chain$try_f.invokeStatic(chain.clj:57)
        at io.pedestal.interceptor.chain$try_f.invoke(chain.clj:44)
        at io.pedestal.interceptor.chain$process_all_with_binding.invokeStatic(chain.clj:171)
        at io.pedestal.interceptor.chain$process_all_with_binding.invoke(chain.clj:146)
        at io.pedestal.interceptor.chain$process_all$fn__31880.invoke(chain.clj:188)
        at clojure.lang.AFn.applyToHelper(AFn.java:152)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.core$apply.invokeStatic(core.clj:646)
        at clojure.core$with_bindings_STAR_.invokeStatic(core.clj:1881)
        at clojure.core$with_bindings_STAR_.doInvoke(core.clj:1881)
        at clojure.lang.RestFn.invoke(RestFn.java:425)
        at io.pedestal.interceptor.chain$process_all.invokeStatic(chain.clj:186)
        at io.pedestal.interceptor.chain$process_all.invoke(chain.clj:182)
        at io.pedestal.interceptor.chain$enter_all.invokeStatic(chain.clj:235)
        at io.pedestal.interceptor.chain$enter_all.invoke(chain.clj:229)
        at io.pedestal.interceptor.chain$execute.invokeStatic(chain.clj:379)
        at io.pedestal.interceptor.chain$execute.invoke(chain.clj:352)
        at com.walmartlabs.lacinia.pedestal.subscriptions$execute_query_interceptors.invokeStatic(subscriptions.clj:78)
        at com.walmartlabs.lacinia.pedestal.subscriptions$execute_query_interceptors.invoke(subscriptions.clj:56)
        at com.walmartlabs.lacinia.pedestal.subscriptions$connection_loop$fn__36575$state_machine__5576__auto____36586$fn__36597.invoke(subscriptions.clj:90)
        at com.walmartlabs.lacinia.pedestal.subscriptions$connection_loop$fn__36575$state_machine__5576__auto____36586.invoke(subscriptions.clj:90)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:973)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:972)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:977)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:975)
        at clojure.core.async$ioc_alts_BANG_$fn__5779.invoke(async.clj:384)
        at clojure.core.async$do_alts$fn__5725$fn__5728.invoke(async.clj:253)
        at clojure.core.async.impl.channels.ManyToManyChannel$fn__447$fn__448.invoke(channels.clj:95)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: clojure.lang.ExceptionInfo: Query validation errors.
        at clojure.core$ex_info.invokeStatic(core.clj:4617)
        at clojure.core$ex_info.invoke(core.clj:4617)
        at com.walmartlabs.lacinia.pedestal.subscriptions$query_parser_interceptor$fn__36790.invoke(subscriptions.clj:234)
        at io.pedestal.interceptor.chain$try_f.invokeStatic(chain.clj:54)
        ... 31 common frames omitted
```

**Comment by hlship** [comment]
I think the right thing here is to send a single websocket message with the response map (containing :errors but not :data).

**Comment by hlship** [comment]
Looking at the code, this is likely going to be fixed in lacinia-pedestal.

**Comment by hlship** [comment]
This will be fixed in lacinia-pedestal 0.6.0.  https://github.com/walmartlabs/lacinia-pedestal/pull/46
